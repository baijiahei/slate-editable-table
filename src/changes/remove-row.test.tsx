import { Editor, Value, ValueJSON } from 'slate';
import { EditTable, TableEditor } from '../index';
import value from '../../mocks/two-by-two';
import { createHtml } from './test-helper';

describe('remove-row', function() {
  const plugin = EditTable();
  it('remove row in 2X2 table', function() {
    const editor = new Editor({ value: Value.fromJSON(value as ValueJSON), plugins: [plugin] });
    const cursorBlock = editor.value.document.getDescendant('0');
    if (!cursorBlock) throw new Error('Failed to find block');
    editor.moveTo(cursorBlock.key);
    const actual = (editor as TableEditor).removeRow().value;
    expect(actual.toJSON()).toMatchSnapshot();
    expect(createHtml(actual)).toMatchSnapshot();
  });
});
