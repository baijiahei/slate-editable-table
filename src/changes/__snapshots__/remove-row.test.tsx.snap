// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remove-row remove row in 2X2 table 1`] = `
Object {
  "document": Object {
    "data": Object {},
    "nodes": Array [
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "marks": Array [],
            "object": "text",
            "text": "Lorem ipsum ...",
          },
        ],
        "object": "block",
        "type": "paragraph",
      },
      Object {
        "data": Object {},
        "nodes": Array [
          Object {
            "data": Object {},
            "nodes": Array [
              Object {
                "data": Object {},
                "nodes": Array [
                  Object {
                    "data": Object {},
                    "nodes": Array [
                      Object {
                        "marks": Array [],
                        "object": "text",
                        "text": "2",
                      },
                    ],
                    "object": "block",
                    "type": "paragraph",
                  },
                ],
                "object": "block",
                "type": "table_cell",
              },
              Object {
                "data": Object {},
                "nodes": Array [
                  Object {
                    "data": Object {},
                    "nodes": Array [
                      Object {
                        "marks": Array [],
                        "object": "text",
                        "text": "3",
                      },
                    ],
                    "object": "block",
                    "type": "paragraph",
                  },
                ],
                "object": "block",
                "type": "table_cell",
              },
            ],
            "object": "block",
            "type": "table_row",
          },
        ],
        "object": "block",
        "type": "table",
      },
    ],
    "object": "document",
  },
  "object": "value",
}
`;

exports[`remove-row remove row in 2X2 table 2`] = `"<p style=\\"margin:0\\">Lorem ipsum ...</p><table><tbody><tr><td style=\\"width:auto\\"><p style=\\"margin:0\\">2</p></td><td style=\\"width:auto\\"><p style=\\"margin:0\\">3</p></td></tr></tbody></table>"`;
